### PR TITLE
Track slugs

### PR DESCRIPTION
**Pull request by angelk**
See https://github.com/trackhub/website/issues/126

### CODE REVIEW

File: translations/validators.bg.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+return [
+    'Should contains only letters, numbers, whitespace, dash and underscore' => 'Трябва да съдържа само букви, числа, дире и долна черта',
```
Comment (StefanMavrodiev) [inline translations/validators.bg.php]: **дире**

File: src/Form/Type/Track.php [hunk 2]
```diff
@@ -84,6 +84,19 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
         );
+
+        $builder->add(
+            'slug',
+            TextType::class,
+            [
+                'required' => false,
+                'label' => 'Short link',
+                'constraints' => new Regex([
+                    'pattern' => '/^[a-zA-Z0-9_\-]+$/',
```
Comment (StefanMavrodiev) [inline src/Form/Type/Track.php]: I don't see whitespace in this pattern.

File: translations/validators.bg.php (hunk 1 above)
Comment (angelk) [reply translations/validators.bg.php]: fixed

File: src/Form/Type/Track.php (hunk 2 above)
Comment (angelk) [reply src/Form/Type/Track.php]: I think that "Momini-skali" looks better than "Momini%20skali" or "Momini+skali"

**Comment by angelk** [comment]
> When the user is accessing page via a slug, all path in the template leads to the track id. You go to
> .../view/foo, make some edits, and the form redirects you to .../view/trackid.

Fixed

File: translations/validators.bg.php [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?php
 
 return [
-    'Should contains only letters, numbers, whitespace, dash and underscore' => 'Трябва да съдържа само букви, числа, дире и долна черта',
+    'Should contains only letters, numbers, whitespace, dash and underscore' => 'Трябва да съдържа само букви, числа, тире и долна черта',
```
Comment (StefanMavrodiev) [inline translations/validators.bg.php]: Please remove **whitespace** to match with translation

Comment (angelk) [reply translations/validators.bg.php]: fixed

**Pull request merged**